toxcore/DHT.c

```diff
@@ -2965,6 +2965,34 @@ bool dht_non_lan_connected(const DHT *dht)
     return false;
 }
 
+uint16_t dht_get_num_closelist(const DHT *dht) {
+    uint16_t num_valid_close_clients = 0;
+    for (size_t i = 0; i < LCLIENT_LIST; ++i) {
+        const Client_data *const client = dht_get_close_client(dht, i);
+
+        // check if client is valid
+        if (!(assoc_timeout(dht->cur_time, &client->assoc4) && assoc_timeout(dht->cur_time, &client->assoc6))) {
+            ++num_valid_close_clients;
+        }
+    }
+
+    return num_valid_close_clients;
+}
+
+uint16_t dht_get_num_closelist_announce_capable(const DHT *dht) {
+    uint16_t num_valid_close_clients_with_cap = 0;
+    for (size_t i = 0; i < LCLIENT_LIST; ++i) {
```

Comment:
uint32_t, because that's what dht_get_close_client expects.

Reply:
right